app/controllers/banners_controller.rb

```diff
@@ -11,18 +11,22 @@ def new
     authorize :application, :admin_or_supervisor?
 
     @banner = Banner.new
+    @has_alternate_active_banner = current_organization.banners.where(active: true).where.not(id: @banner.id).exists?
   end
 
   def edit
     authorize :application, :admin_or_supervisor?
 
     @banner = current_organization.banners.find(params[:id])
+    @has_alternate_active_banner = current_organization.banners.where(active: true).where.not(id: @banner.id).exists?
   end
 
   def create
     authorize :application, :admin_or_supervisor?
 
     @banner = current_organization.banners.build(banner_params)
+    deactivate_alternate_active_banner if banner_params[:active]&.to_i == 1
+
```

Comment:
Wanted to avoid repeating this in two methods and tried with a `:before_action, only: [:create, :edit]`, but didn't know how to ensure we had an `@banner` to work with.

Reply:
I think its fine to double up the code. Create and update often have similar code.